src/main/java/com/opentable/kafka/util/OffsetMetrics.java

```diff
@@ -0,0 +1,183 @@
+package com.opentable.kafka.util;
+
+import java.io.Closeable;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricSet;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.opentable.concurrent.OTExecutors;
+import com.opentable.metrics.AtomicLongGauge;
+
+/**
+ * {@link MetricSet} implementation that monitors consumer groups consuming from topics.
+ *
+ * Runs a thread that polls the Kafka broker; use {@link #start()} and {@link #stop()} to spin up and down.
+ *
+ * The metric namespace is as follows.
+ *
+ * <p>
+ * {@code <topic>.{partition.%d,total}.{size,offset,lag}}
+ *
+ * <p>
+ * Size is the maximum offset of the topic (this is independent of any consumer). Offset is the current offset of the
+ * consumer. Lag is the size minus the offset. Totals are summed across the partitions.
+ */
+public class OffsetMetrics implements MetricSet, AutoCloseable, Closeable {
```

Comment:
autocloseable is redundant

Reply:
Oh, you're right.  Thanks!  Implemented in 0c68d98f4d6bf4258f909fbb2dca18ca31a39593.